### DISCUSSION TITLE
double Integrate Postprocessor

### DISCUSSION THREAD

**Discussion by bosxered** [idaholab/moose#24564] [Q&A General] [answered]
Hello Moose Experts.

I found a postprocessor called SideIntegralVariablePostprocessor that integrates variable u over the side.

In a 2D system, it calculates, at the bottom, 
![image](https://github.com/idaholab/moose/assets/79071957/6acbb485-1eb9-4c5f-a646-6133ead46854)

I want to calculate 
![image](https://github.com/idaholab/moose/assets/79071957/adc94446-cc32-4e0c-b908-49c0a4f8c104)

Is there any postprocessor that calculates the above one?

Thank you.

Best wishes,
Jeonghwan

**Comment by GiudGiud** [comment]
Hello

What are you trying to compute? This does not seem right, a "side" of a 2D system is 1D so the differential element is dx not dx*dx.
Are you looking for a volumetric (2d element in 2D) integration?

To your question, no there is not. But you could make your own by modifying the SideIntegralPostprocessor which has the quadrature integration calculation

GUillaume

**Reply by bosxered** [reply]
![image](https://github.com/idaholab/moose/assets/79071957/91cda315-8aa8-48ba-842e-177c8ab359c5)

I am trying to simulate a 2D system assuming a cylindrical symmetry just like the above picture.
I have to calculate the electrical current using the current density at the bottom.
I think `Sigma(u*3.14*dx*dx)` is reasonable because I am assuming cylindrical symmetry.

Doesn't it make sense?

Actually, I am a beginner in C++ language and I have not sure that I can modify well. That is why I was finding an already-implemented postprocessor. Could you check if I am doing right?

From the SideIntegralPostprocessor.C, I modified "::computeIntegral()" as follows

```
Real
SideIntegralPostprocessor::computeIntegral()
{
  Real sum = 0;
  if (_qp_integration)
    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
      sum += _JxW[_qp] * _coord[_qp] * _coord[_qp] * computeQpIntegral();
  else
  {
    // Finite volume functors integration is over FaceInfo, not quadrature points
    getFaceInfos();

    for (auto & fi : _face_infos)
      sum += fi->faceArea() * fi->faceCoord() * computeFaceInfoIntegral(fi);
  }
  return sum;
}
```
The modified line is just `sum += _JxW[_qp] * _coord[_qp] * _coord[_qp] * computeQpIntegral();`
and I am trying to make a new postprocessor "SideDoubleIntegralValuePostprocessor" from this "SideDoubleIntegralPostprocessor".
Is it make sense?

Thank you

**Reply by GiudGiud** [reply]
I still dont think you need to do this.

The _coord in RZ is here to multiply by the radius
JxW has the local weight of the quadrature and the mapping between elements

Are you performing RZ simulations or 3D with a cylinder system?

**Reply by bosxered** [reply]
No. I am performing a 2D simulation (x,y).
The red box is my system.
![image](https://github.com/idaholab/moose/assets/79071957/78d2be6f-c57c-4574-ad5e-fb0b0c36f42c)


And this is my Exodus output results of J which is the current density.
![image](https://github.com/idaholab/moose/assets/79071957/6f5f46ee-da03-47f8-bfe3-8814fcf40c48)


I am performing 2D simulation but if I assume this has cylindrical symmetry, I can consider this as a 3D cylinder system by just doing 2D simulation.
As I mentioned, I want to calculate the current(A) from the current density(A/m^2).
**Even if I am doing a 2D simulation for computational efficiency, I want to get the current value in a 3D system with simple and reasonable assumptions.**
To do that, I want to calculate (in this case, u is the current density)
![image](https://github.com/idaholab/moose/assets/79071957/f3d0f860-5f10-410d-a4e0-c68e885592ec)
this one because, if it is right, I can get the current value by multiplying 3.14.

For example,
![image](https://github.com/idaholab/moose/assets/79071957/793bbbff-926d-4721-9ec8-26183926f83c)

If the J is the same over the boundary just like the above simple example, I can just use SideaveragePostprocessor and multiply using Parsedpostprocessor.
Unfortunately, it isn't so that I cannot compare with current values from different systems.

**So, does your answer mean that my modification of the code is correct?**

Thank you.

**Reply by GiudGiud** [reply]
Are you using coord_type = RZ in the Mesh block?

This multiplication to go from the bottom line to the blue area should be already performed by a SideIntegralPostprcoessor if you are using coord_type = RZ

**Reply by bosxered** [reply]
No. I am not using RZ coord_type.

My simulated system is only a 2D green rectangular system.
Using results from the rectangular system, I can get 3D system results without 3D simulation when I can calculate 
![image](https://github.com/idaholab/moose/assets/79071957/5326028a-63f6-4206-b5e9-22b8468ed32f)
because my 2D system has cylindrical symmetry.

**Reply by GiudGiud** [reply, marked as answer]
Well there's your problem.
If your 2D system has cylindrical symmetry, you have to specify the coordinate type of RZ in the Mesh block.
`coord_type = RZ`

If you dont, the postprocessing is wrong, but the solve as well. All the kernels are acting on elements of the wrong size.

**Reply by bosxered** [reply]
Okay.
I will try the corrd_type = RZ.
Thank you very much!!
